test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/ArraysTest.scala

```diff
@@ -519,6 +519,50 @@ class ArraysTest {
     assertEquals(-7, ret)
   }
 
+  @Test def binarySearchWithStartAndEndIndexOnSpecificAnyRefWithComparator(): Unit = {
+    val cmp = new java.util.Comparator[(Int, Int)] {
```

Comment:
We might want to start using the factories on `Comparator` for this in another PR (and implement if we don't have them).

```
Comparator.comparing(_._1).thenComparing(_._2) // type inference probably can't deal with this :-/
```

Probably for consistency with the rest of the file, its better to leave it like this though.